### PR TITLE
Added option to set custom video size.

### PR DESCRIPTION
**Pull request by robderijk**

### CODE REVIEW

File: ios/Classes/CameraPreview/CameraPreview.m [hunk 1]
```diff
@@ -113,7 +114,19 @@ - (void)setCameraPresset:(CGSize)currentPreviewSize {
     }
     [_captureSession setSessionPreset:presetSelected];
     _currentPresset = presetSelected;
-    
+}
+
+/// Set camera video size
+- (void)setCameraVideoPresset:(CGSize)currentPreviewSize {
```
Comment (sdgroot) [inline ios/Classes/CameraPreview/CameraPreview.m]: This should be named `setCameraPreviewPresset`, and the other one should have this name I think. Also `currentPreviewSize` is misleading, because it is the new/requested size.

```suggestion
- (void)setCameraPreviewPreset:(CGSize)requestedSize {
```

File: ios/Classes/CameraPreview/CameraPreview.m [hunk 2]
```diff
@@ -102,7 +103,7 @@ - (void)dealloc {
 }
 
 /// Set camera preview size
-- (void)setCameraPresset:(CGSize)currentPreviewSize {
+- (void)setCameraPreviewPresset:(CGSize)currentPreviewSize {
```
Comment (sdgroot) [inline ios/Classes/CameraPreview/CameraPreview.m:106]: Why is this renamed to `setCameraPreviewPresset`? It sets the video output preset, not the preview preset.

```suggestion
- (void)setCameraVideoPreset:(CGSize)requestedSize {
```

File: lib/camerapreview.dart [hunk 3]
```diff
@@ -379,6 +393,13 @@ class CameraAwesomeState extends State<CameraAwesome> with WidgetsBindingObserve
     });
   }
 
+  _initVideoSize() {
+    widget.videoSize.addListener(() async {
+      await CamerawesomePlugin.setVideoSize(
+          widget.videoSize.value.width.toInt(), widget.videoSize.value.height.toInt());
+    });
+  }
```
Comment (sdgroot) [inline lib/camerapreview.dart:401]: 1. Should this listener be removed somewhere in the lifecycle?
2. Is this listener really required? It looks to me that the video size is only set once, so it may be less error prone to just update the video size there immediately.

**Review by sdgroot — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/apparence/camerawesome/CameraPicture.java [hunk 4]
```diff
@@ -81,41 +81,61 @@ public CameraPicture(Context context, CameraPreview cameraPreview, CameraSession
      * @param width
      * @param height
      */
-    public void setSize(int width, int height) {
-        this.size = new Size(width, height);
+    public void setPhotoSize(int width, int height) {
+        this.photoSize = new Size(width, height);
```
Comment (wimwakel) [inline android/src/main/java/com/apparence/camerawesome/CameraPicture.java:85]: should photoSize and videoSize be unset when either of them is set? 

(i don't know the lifecycle of the CameraPicture)

File: ios/Classes/CameraPreview/CameraPreview.m (hunk 2 above)
Comment (robderijk) [reply ios/Classes/CameraPreview/CameraPreview.m:106]: Because it sets the preview preset, not the video output preset: `[_captureSession setSessionPreset:presetSelected];` is the preview.
I assume you're confused because `[CameraQualities selectVideoCapturePresset:currentPreviewSize session:_captureSession];` is used, that's just bad naming by the library, inside the implementation of that method nothing is specific to video.

File: ios/Classes/CameraPreview/CameraPreview.m (hunk 1 above)
Comment (robderijk) [reply ios/Classes/CameraPreview/CameraPreview.m]: I updated the code to fix naming confusion and moved the assignment of _currentPreviewSize to the other method. I think this should be alright now, can you re-review?

**Review by sdgroot — APPROVED** [review]

File: android/src/main/java/com/apparence/camerawesome/CameraPicture.java (hunk 4 above)
Comment (robderijk) [reply android/src/main/java/com/apparence/camerawesome/CameraPicture.java:85]: Each time a new CameraAwesome widget is loaded a new CameraPicture object is created so this won't cause issues.
